plugin-registration.md

```diff
@@ -243,6 +243,20 @@ When you register the back-end navigation you can use [localization strings](loc
 
 To make the sub-menu items visible, you may [set the navigation context](../backend/controllers-ajax#navigation-context) in the back-end controller using the `BackendMenu::setContext` method. This will make the parent menu item active and display the children in the side menu.
 
+Key | Description
+------------- | -------------
+**code** | a string value that acts as an unique identifier for that menu option, required.
```

Comment:
While this is a property on the MenuItem class, I don't think you can specify it in the array configuration, since it's the key of the configuration.

Reply:
@LukeTowers 

You've marked both the `code` properties and say they don't come through as an array. But when we print out the following api `BackendMenu::listSideMenuItems();` we see them in the array and get the following result:

```php
Array
(
    [example] => Backend\Classes\SideMenuItem Object
    (
        [code] => example
        [owner] => Acme.Plugin
        [label] => acme.plugin::lang.menu.example.title
        [icon] => fas fa-users-cog
        [iconSvg] =>
        [url] => https://www.example.com/acme/plugin/example
        [counter] =>
        [counterLabel] =>
        [badge] =>
        [order] => 30
        [attributes] => Array
    (
    )
        [permissions] => Array
        (
            [0] => acme.plugin.example
        )
    )
)
```

I thought the doc's should just explain all the properties in the array.

Please advise what to do to resolve your comments.

Reply:
I see what's confused me, when you output the array `BackendMenu::listSideMenuItems();` it shows `code` in the array. But when people write the php code for the sidebar menu in the `Pluygins.php` file , you don't need to write the `'code' => 'example'` so I will remove both of them.

Reply:
It can still be documented here, it just needs to have a disclaimer on where a developer would provide that value.

Reply:
I see, so I'm writing a commit like this:

```
**code** | **this item is auto generated by the cms and should not be added by the developer**, a string value that acts as an unique identifier for that menu option.
**owner** | **this item is auto generated by the cms and should not be added by the developer**, a string value that specifies the menu items owner plugin or module in the format "Author.Plugin".
```